subsquid/src/bondedFinanceProcessor.ts

```diff
@@ -0,0 +1,103 @@
+import { EventHandlerContext } from "@subsquid/substrate-processor";
+import {
+  BondedFinanceNewBondEvent,
+  BondedFinanceNewOfferEvent,
+} from "./types/events";
+import { BondedFinanceBondOffer } from "./model";
+
+interface NewOfferEvent {
+  offerId: bigint;
+}
+
+interface NewBondEvent {
+  offerId: bigint;
+  nbOfBonds: bigint;
+}
+
+/**
+ * Extracts relevant information about a new bond
+ * @param event
+ */
+function getNewBondEvent(event: BondedFinanceNewBondEvent): NewBondEvent {
+  if (event.isV2300) {
+    const { offerId, nbOfBonds } = event.asV2300;
+    return { offerId, nbOfBonds };
+  }
+
+  const { offerId, nbOfBonds } = event.asLatest;
+  return { offerId, nbOfBonds };
+}
+
+function getNewOfferEvent(event: BondedFinanceNewOfferEvent): NewOfferEvent {
+  if (event.isV2300) {
+    const { offerId } = event.asV2300;
+
+    return { offerId };
+  }
+
+  const { offerId } = event.asLatest;
+  return { offerId };
+}
+
+/**
+ * Extracts beneficiary from a bond offer
+ * @param ctx
+ */
+function getBeneficiary(ctx: EventHandlerContext): string {
+  const { beneficiary } = ctx.event.extrinsic?.args[0]?.value as Record<
+    string,
+    unknown
+  >;
+
+  return beneficiary as string;
+}
+
+export async function processNewOfferEvent(
```

Comment:
Why do we need to process this event?

Reply:
Then I propose having another entity to track the bond offers without the attachment to beneficiary

Reply:
We can do that but would there be any benefit of having the beneficiaries on a separate entity, given that the beneficiary does not change over time?

Reply:
Seems with this task we want to track both bond offers and bonds, which have a one to many relationship. With current structure it is not tracked correctly

Reply:
@easteregg

Reply:
Individual bonds are going to play a role once VestingSchedules are trackable. 

Ultimately we would want to get access to the user's individual bond, and its claimable amount, and claim it. 

This might be far from fetching right now, but for now, if we could only get the Total Purchased amount for each offer, that's enough.

Reply:
Cool, lets find out the exact requirement first. Otherwise current structure is ok as long as the need is to track only the fields coming from `BondOffer` entity in the runtime. We can add the beneficiary field to the `NewOffer` event also in the runtime so that a party which only needs beneficiary does not need to process `NewBond` event